asapdiscovery-data/asapdiscovery/data/fragalysis.py

```diff
@@ -170,10 +170,15 @@ def parse_fragalysis(
         df = df.drop_duplicates("RealCrystalName")
 
     # Build argument dicts for the CrystalCompoundData objects
-    xtal_dicts = [
-        dict(zip(("smiles", "dataset", "compound_id"), r[1].values))
-        for r in df.loc[:, ["smiles", "crystal_name", "alternate_name"]].iterrows()
-    ]
+    try:
+        xtal_dicts = [
+            dict(zip(("smiles", "dataset", "compound_id"), r[1].values))
+            for r in df.loc[:, ["smiles", "crystal_name", "alternate_name"]].iterrows()
+        ]
+    except KeyError as e:
+        raise Exception(
+            "Did you use 'Mpro_compound_tracker_csv.csv'? Use 'metadata.csv' instead."
```

Comment:
Are these files being provided by our package? If not, then we probably need to specify what the actual format it expects (keys and values) rather than the names of the files.

Reply:
They are Fragalysis-generated files. And since this is a module intended for processing the Fragalysis database I think this make sense?

Reply:
I'm just saying that for me this exception message doesn't say much. Are users expected to know what the file names mean and how they are different for years to come? And if we know how they are different maybe that's what we want to print in this message (assuming it can be put in a few words).

Reply:
fair enough. added to the message!